In [3]:
import pandas as pd
import numpy as np
import openpyxl
pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)

In [4]:
# Identify Sheet Names in the File

filePath = r"D:\Astragraphia\BO1\Axioo\Assessment Data\Copy of Control Asset Printer Maret Mei 2023.xlsx"
daFile = openpyxl.load_workbook(filePath)
daSheet = daFile.sheetnames
daFile.sheetnames

['List Printer Office',
 'List Printer Gudang Produksi',
 'Report Perbaikan Printer Maret',
 'Report Perbaikan Printer April']

In [18]:
daList = pd.read_excel(
    filePath,
    sheet_name = daSheet[0]
)
daList.head()

Penggunaan printer di TDI                           Unnamed: 1  Unnamed: 2  \
0                       NaN                                  NaN         NaN   
1                        No                               Lokasi  Department   
2                         1       Office lt.1 Ruang CS IT Clinic   IT Clinic   
3                         2  Office lt.1 Ruang Teknisi IT Clinic   IT Clinic   
4                         3            Office Lt.2 Ruang Project     Project   

     Unnamed: 3              Unnamed: 4                  Unnamed: 5  \
0           NaN                     NaN                         NaN   
1  Merk Printer            Type Printer  Pemakaian 1 bulan (lembar)   
2           HP    107 A Black and White                        1000   
3        Epson   LX 300 Black and white                         500   
4         Canon             G3010 Color                         200   

                                          Unnamed: 6  \
0                                                NaN   
1  Penggantian part/toner dalam 1 bulan \n(Maret ...   
2                                                NaN   
3                                                NaN   
4                                                NaN   

                                          Unnamed: 7  
0                                                NaN  
1  Penggantian part/toner dalam 1 bulan \n(April ...  
2                                                NaN  
3                                                NaN  
4                                                NaN

In [45]:
#daList = daList.iloc[1:len(daList)-1]
daList.head()
try:
    print(list(daList[daList.columns[1]].iloc[0:1])[0])
except Exception as e:
    print(e)

dOld = list(daList.columns)
dNew = []
for a in range(0,len(dOld)):
    dNew.append(list(daList[dOld[a]].iloc[0:1])[0])

Lokasi


In [48]:
daList.head()

Penggunaan printer di TDI                           Unnamed: 1  Unnamed: 2  \
1                        No                               Lokasi  Department   
2                         1       Office lt.1 Ruang CS IT Clinic   IT Clinic   
3                         2  Office lt.1 Ruang Teknisi IT Clinic   IT Clinic   
4                         3            Office Lt.2 Ruang Project     Project   
5                         4              Office lt.2 Ruang Legal       Legal   

     Unnamed: 3              Unnamed: 4                  Unnamed: 5  \
1  Merk Printer            Type Printer  Pemakaian 1 bulan (lembar)   
2           HP    107 A Black and White                        1000   
3        Epson   LX 300 Black and white                         500   
4         Canon             G3010 Color                         200   
5           HP    Office Jet 7720 Color                         300   

                                          Unnamed: 6  \
1  Penggantian part/toner dalam 1 bulan \n(Maret ...   
2                                                NaN   
3                                                NaN   
4                                                NaN   
5                                                NaN   

                                          Unnamed: 7  
1  Penggantian part/toner dalam 1 bulan \n(April ...  
2                                                NaN  
3                                                NaN  
4                                                NaN  
5                                                NaN

In [51]:
print(dNew)
print(dOld)

for o, n in zip(dOld, dNew):
    daList.rename(columns={o:n}, inplace = True)
daList.drop(index = 1, inplace = True)

['No', 'Lokasi', 'Department', 'Merk Printer', 'Type Printer', 'Pemakaian 1 bulan (lembar)', 'Penggantian part/toner dalam 1 bulan \n(Maret 2024)', 'Penggantian part/toner dalam 1 bulan \n(April 2024)']
['Penggunaan printer di TDI', 'Unnamed: 1', 'Unnamed: 2', 'Unnamed: 3', 'Unnamed: 4', 'Unnamed: 5', 'Unnamed: 6', 'Unnamed: 7']


NameError: name 'dList' is not defined

In [63]:
#daList = daList.reset_index()
#daList.drop(columns = ['index'], inplace = True)
d = list(daList.columns)
daList[[d[1],d[2]]].fillna(method = 'ffill', inplace = True)
daList.to_excel("D:\\Astragraphia\\BO1\\Axioo\\Assessment Data\\Asset Printer Maret Mei 2023.xlsx")
daList.to_csv("D:\\Astragraphia\\BO1\\Axioo\\Assessment Data\\Asset Printer Maret Mei 2023.csv")

C:\Users\Garimahdi.Supandi\AppData\Local\Temp\ipykernel_16492\1461803249.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  daList[[d[1],d[2]]].fillna(method = 'ffill', inplace = True)


# Just load from here

In [2]:
axioo = pd.read_csv(
    r"D:\\Astragraphia\\BO1\\Axioo\\Assessment Data\\Asset Printer Maret Mei 2023.csv",
    encoding = 'utf-8'
)

In [3]:
c = list(axioo.columns)
axioo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32 entries, 0 to 31
Data columns (total 9 columns):
 #   Column                                              Non-Null Count  Dtype 
---  ------                                              --------------  ----- 
 0   Unnamed: 0                                          32 non-null     int64 
 1   No                                                  32 non-null     int64 
 2   Lokasi                                              15 non-null     object
 3   Department                                          11 non-null     object
 4   Merk Printer                                        32 non-null     object
 5   Type Printer                                        32 non-null     object
 6   Pemakaian 1 bulan (lembar)                          32 non-null     int64 
 7   Penggantian part/toner dalam 1 bulan 
(Maret 2024)  3 non-null      object
 8   Penggantian part/toner dalam 1 bulan 
(April 2024)  5 non-null      object
dtypes: int64(3), 

In [4]:
axioo.head()

Unnamed: 0  No                               Lokasi Department  \
0           0   1       Office lt.1 Ruang CS IT Clinic  IT Clinic   
1           1   2  Office lt.1 Ruang Teknisi IT Clinic  IT Clinic   
2           2   3            Office Lt.2 Ruang Project    Project   
3           3   4              Office lt.2 Ruang Legal      Legal   
4           4   5                 Office lt.2 Ruang HR         HR   

  Merk Printer            Type Printer  Pemakaian 1 bulan (lembar)  \
0          HP    107 A Black and White                        1000   
1       Epson   LX 300 Black and white                         500   
2        Canon             G3010 Color                         200   
3          HP    Office Jet 7720 Color                         300   
4        Canon           IP 2770 Color                         100   

  Penggantian part/toner dalam 1 bulan \n(Maret 2024)  \
0                                                NaN    
1                                                NaN    
2                                                NaN    
3                                                NaN    
4                                                NaN    

  Penggantian part/toner dalam 1 bulan \n(April 2024)  
0                                                NaN   
1                                                NaN   
2                                                NaN   
3                                                NaN   
4                                                NaN

In [5]:
axioo.drop(columns=c[0], inplace = True)
c = list(axioo.columns)

In [6]:
axioo[c[1]].fillna(method = 'ffill', inplace = True)
axioo[c[2]].fillna(method = 'ffill', inplace = True)

In [7]:
axioo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32 entries, 0 to 31
Data columns (total 8 columns):
 #   Column                                              Non-Null Count  Dtype 
---  ------                                              --------------  ----- 
 0   No                                                  32 non-null     int64 
 1   Lokasi                                              32 non-null     object
 2   Department                                          32 non-null     object
 3   Merk Printer                                        32 non-null     object
 4   Type Printer                                        32 non-null     object
 5   Pemakaian 1 bulan (lembar)                          32 non-null     int64 
 6   Penggantian part/toner dalam 1 bulan 
(Maret 2024)  3 non-null      object
 7   Penggantian part/toner dalam 1 bulan 
(April 2024)  5 non-null      object
dtypes: int64(2), object(6)
memory usage: 2.1+ KB


In [9]:
axioo.to_csv(
    "D:\\Astragraphia\\BO1\\Axioo\\Assessment Data\\Asset Printer Maret Mei 2023.csv",
    encoding = 'utf-8',
    index = False
)

In [10]:
axioo = pd.read_csv(
    r"D:\\Astragraphia\\BO1\\Axioo\\Assessment Data\\Asset Printer Maret Mei 2023.csv",
    encoding = 'utf-8'
)
c = list(axioo.columns)

In [32]:
axioo.info()
no_nan = axioo.loc[pd.notna(axioo[c[9]])]
no_nan.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32 entries, 0 to 31
Data columns (total 10 columns):
 #   Column                                              Non-Null Count  Dtype  
---  ------                                              --------------  -----  
 0   No                                                  32 non-null     int64  
 1   Lokasi                                              32 non-null     object 
 2   Department                                          32 non-null     object 
 3   Merk Printer                                        32 non-null     object 
 4   Type Printer                                        32 non-null     object 
 5   Pemakaian 1 bulan (lembar)                          32 non-null     int64  
 6   Penggantian part/toner dalam 1 bulan 
(Maret 2024)  3 non-null      object 
 7   Penggantian part/toner dalam 1 bulan 
(April 2024)  5 non-null      object 
 8   CLR Speed (ppm)                                     21 non-null     float64
 9   B

In [31]:
no_nan.to_csv(
    "D:\\Astragraphia\\BO1\\Axioo\\Assessment Data\\Axioo Maret Mei 2023.csv",
    encoding = 'utf-8',
    index = False
)

no_nan.to_excel(
    "D:\\Astragraphia\\BO1\\Axioo\\Assessment Data\\Axioo Maret Mei 2023.xlsx",
    index = False
)

In [3]:
axioo = pd.read_csv(
    r"D:\\Astragraphia\\BO1\\Axioo\\Assessment Data\\Axioo Maret Mei 2023.csv",
    encoding = 'utf-8'
)
c = list(axioo.columns)
print(axioo.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21 entries, 0 to 20
Data columns (total 10 columns):
 #   Column                                              Non-Null Count  Dtype  
---  ------                                              --------------  -----  
 0   No                                                  21 non-null     int64  
 1   Lokasi                                              21 non-null     object 
 2   Department                                          21 non-null     object 
 3   Merk Printer                                        21 non-null     object 
 4   Type Printer                                        21 non-null     object 
 5   Pemakaian 1 bulan (lembar)                          21 non-null     int64  
 6   Penggantian part/toner dalam 1 bulan 
(Maret 2024)  2 non-null      object 
 7   Penggantian part/toner dalam 1 bulan 
(April 2024)  5 non-null      object 
 8   CLR Speed (ppm)                                     21 non-null     float64
 9   B

In [18]:
axioo['Min Utilisation (%)'] = ((axioo[c[5]])/(axioo[c[8]]*60*8*23))*100
axioo['Max Utilisation (%)'] = ((axioo[c[5]])/(axioo[c[9]]*60*8*23))*100
axioo['Min Utilisation (%)'].replace({np.inf:0}, inplace = True)

In [20]:
axioo.head()
axioo['Min Utilisation (%)'] = axioo['Min Utilisation (%)'].apply(lambda x:round(x,2))
axioo['Max Utilisation (%)'] = axioo['Max Utilisation (%)'].apply(lambda x:round(x,2))

In [21]:
axioo.head()

No                          Lokasi Department Merk Printer  \
0   1  Office lt.1 Ruang CS IT Clinic  IT Clinic          HP    
1   3       Office Lt.2 Ruang Project    Project        Canon   
2   4         Office lt.2 Ruang Legal      Legal          HP    
3   5            Office lt.2 Ruang HR         HR        Canon   
4   6            Office lt. Ruang SCM        SCM          HP    

            Type Printer  Pemakaian 1 bulan (lembar)  \
0  107 A Black and White                        1000   
1            G3010 Color                         200   
2  Office Jet 7720 Color                         300   
3          IP 2770 Color                         100   
4          M183 FW Color                        6000   

  Penggantian part/toner dalam 1 bulan \n(Maret 2024)  \
0                                                NaN    
1                                                NaN    
2                                                NaN    
3                                                NaN    
4             03/05/2024 Refil tinta warna dan hitam    

  Penggantian part/toner dalam 1 bulan \n(April 2024)  CLR Speed (ppm)  \
0                                                NaN               0.0   
1                                                NaN               5.0   
2                                                NaN              18.0   
3                                                NaN               4.8   
4                                                NaN              17.0   

   BW Speed (ppm)  Min Utilisation (%)  Max Utilisation (%)  
0            21.0                 0.00                 0.43  
1             8.8                 0.36                 0.21  
2            22.0                 0.15                 0.12  
3             7.0                 0.19                 0.13  
4            17.0                 3.20                 3.20

In [22]:
c = list(axioo.columns)
axioo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21 entries, 0 to 20
Data columns (total 12 columns):
 #   Column                                              Non-Null Count  Dtype  
---  ------                                              --------------  -----  
 0   No                                                  21 non-null     int64  
 1   Lokasi                                              21 non-null     object 
 2   Department                                          21 non-null     object 
 3   Merk Printer                                        21 non-null     object 
 4   Type Printer                                        21 non-null     object 
 5   Pemakaian 1 bulan (lembar)                          21 non-null     int64  
 6   Penggantian part/toner dalam 1 bulan 
(Maret 2024)  2 non-null      object 
 7   Penggantian part/toner dalam 1 bulan 
(April 2024)  5 non-null      object 
 8   CLR Speed (ppm)                                     21 non-null     float64
 9   B

In [28]:
print(axioo[c[10]].unique(),'\n')
print(axioo[c[11]].unique())
axioo.drop(columns = c[0], inplace = True)
axioo.to_csv(
    "D:\\Astragraphia\\BO1\\Axioo\\Assessment Data\\Axioo Maret Mei 2023.csv",
    encoding = 'utf-8',
    index = False
)

axioo.to_excel(
    "D:\\Astragraphia\\BO1\\Axioo\\Assessment Data\\Axioo Maret Mei 2023.xlsx",
    index = False
)

[0.   0.36 0.15 0.19 3.2  2.01 3.02 1.74 0.12 5.43] 

[0.43 0.21 0.12 0.13 3.2  1.01 0.09 1.91 1.25 0.05 2.59 0.69 0.6  0.3
 0.41 0.52 0.32 0.22]


Darth vader

Mop
Bw: 50
Clr: 150
Breakdown: None, suka nyangkut
Ganti: 3 bln 1x
User: 3 org
Agak susah mondar mandir

Kedalem dikit:

Epson L4260
Mop 600
Bw: 0.2x
Clr: 0.8x

Hp pro m127n
Mop
Dominan scan
Lgsg msk nb scannya
Breakdown seminggu, kalo tinta doang sehari
Sebulan 1x buat bw, 1x 3 bln buat clr

10 org user



2770 QA
100
Bw: 0.2x
Clr: 0.8x
Breakdown seminggu, worst sebulan, avg seminggu
Consume sebulan 1x
5 org user

Office room warehouse

107a
Bw: 500
Clr: -

Consume 3 bln sekali
User 6 org
Breakdown less 24h
Consume distok

Sebelahnya
107a
Bw: 1000
Breakdown seminggu
Consume 3 bln 1x
User 1 org



Gudang lt2
107a hp dpn
Bw 1000
4 bln lbh
1 org

M1100 blkg
Bw 1000
Breakdown sebulan
Tinta refill, 4 bln 1x
3 org

Warehouse beneran

107a
500 bw
4x sebulan
1 org pake



107a
500 bw
4x sebulan
1 org pake

In [2]:
axioo = pd.read_csv(
    r"D:\\Astragraphia\\BO1\\Axioo\\Assessment Data\\Axioo Maret Mei 2023.csv",
    encoding = 'utf-8'
)

axioo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21 entries, 0 to 20
Data columns (total 11 columns):
 #   Column                                              Non-Null Count  Dtype  
---  ------                                              --------------  -----  
 0   Lokasi                                              21 non-null     object 
 1   Department                                          21 non-null     object 
 2   Merk Printer                                        21 non-null     object 
 3   Type Printer                                        21 non-null     object 
 4   Pemakaian 1 bulan (lembar)                          21 non-null     int64  
 5   Penggantian part/toner dalam 1 bulan 
(Maret 2024)  2 non-null      object 
 6   Penggantian part/toner dalam 1 bulan 
(April 2024)  5 non-null      object 
 7   CLR Speed (ppm)                                     21 non-null     float64
 8   BW Speed (ppm)                                      21 non-null     float64
 9   M

In [3]:
a = list(axioo.columns)
axioo[a[4]].sum()

30600

In [9]:
mos = [22,22,22,22,22,23,23,23,23,23,23,23]
lo = (len(mos)//2)-1
hi = (len(mos)//2)+1
median = (mos[lo]+mos[hi])/2
median
avg = round((sum(mos))/len(mos))
avg

23

In [4]:
axioo = pd.read_csv(
    r"D:\\Astragraphia\\BO1\\Axioo\\Assessment Data\\Axioo Maret Mei 2023.csv",
    encoding = 'utf-8'
)

c = list(axioo.columns)
axioo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21 entries, 0 to 20
Data columns (total 11 columns):
 #   Column                                              Non-Null Count  Dtype  
---  ------                                              --------------  -----  
 0   Lokasi                                              21 non-null     object 
 1   Department                                          21 non-null     object 
 2   Merk Printer                                        21 non-null     object 
 3   Type Printer                                        21 non-null     object 
 4   Pemakaian 1 bulan (lembar)                          21 non-null     int64  
 5   Penggantian part/toner dalam 1 bulan 
(Maret 2024)  2 non-null      object 
 6   Penggantian part/toner dalam 1 bulan 
(April 2024)  5 non-null      object 
 7   CLR Speed (ppm)                                     21 non-null     float64
 8   BW Speed (ppm)                                      21 non-null     float64
 9   M

In [5]:
axioo.head()

Lokasi Department Merk Printer  \
0  Office lt.1 Ruang CS IT Clinic  IT Clinic          HP    
1       Office Lt.2 Ruang Project    Project        Canon   
2         Office lt.2 Ruang Legal      Legal          HP    
3            Office lt.2 Ruang HR         HR        Canon   
4            Office lt. Ruang SCM        SCM          HP    

            Type Printer  Pemakaian 1 bulan (lembar)  \
0  107 A Black and White                        1000   
1            G3010 Color                         200   
2  Office Jet 7720 Color                         300   
3          IP 2770 Color                         100   
4          M183 FW Color                        6000   

  Penggantian part/toner dalam 1 bulan \n(Maret 2024)  \
0                                                NaN    
1                                                NaN    
2                                                NaN    
3                                                NaN    
4             03/05/2024 Refil tinta warna dan hitam    

  Penggantian part/toner dalam 1 bulan \n(April 2024)  CLR Speed (ppm)  \
0                                                NaN               0.0   
1                                                NaN               5.0   
2                                                NaN              18.0   
3                                                NaN               4.8   
4                                                NaN              17.0   

   BW Speed (ppm)  Min Utilisation (%)  Max Utilisation (%)  
0            21.0                 0.00                 0.43  
1             8.8                 0.36                 0.21  
2            22.0                 0.15                 0.12  
3             7.0                 0.19                 0.13  
4            17.0                 3.20                 3.20

In [8]:
axioo['C Yield'] = None
axioo['M Yield'] = None
axioo['Y Yield'] = None
axioo['K Yield'] = None
axioo['C Cost'] = None
axioo['M Cost'] = None
axioo['Y Cost'] = None
axioo['K Cost'] = None
util_yield = axioo[[c[0],c[2],c[3],c[8],'C Yield','M Yield','Y Yield','K Yield',c[4],c[10]]]
toner_cost = axioo[[c[0],c[2],c[3],'C Cost','M Cost','Y Cost','K Cost']]
sitrep = axioo[[c[0],c[2],c[3],c[5],c[6]]]

In [16]:
util_yield.to_excel(
    "D:\\Astragraphia\\BO1\\Axioo\\Assessment Data\\axioo_assessment_rev.xlsx",
    index = False
)

In [17]:
path = r"D:\\Astragraphia\\BO1\\Axioo\\Assessment Data\\axioo_assessment_rev.xlsx"
with pd.ExcelWriter(path, engine='openpyxl') as writer:
    util_yield.to_excel(writer, sheet_name='Util Yield', index=False)
    toner_cost.to_excel(writer, sheet_name='Toner-Other Costs', index=False)
    sitrep.to_excel(writer, sheet_name='Devices SitRep', index=False)